src/language/i18n.js

```diff
@@ -141,6 +139,10 @@ const LanguageProvider = ({ children }) => {
   const setLanguage = useCallback(
     async language => {
       if (!language) {
+        // if no language is passed, setLanguage is called by user through the device default setting
```

Comment:
couldn't we detect sys locale by I18n.getInitialLocale() instead of restart ?

```
const setLanguage = useCallback(
     async language => {
      let locale = language

       if (!language) {
         await AsyncStorage.removeItem('lang')
         locale = await I18n.getInitialLocale()
       }

      await I18n.dynamicActivate(language)
      setCurrentLanguage(language)
    },
    [setCurrentLanguage],
  )  
```